src/app/options/qgsuserprofileselectiondialog.cpp

```diff
@@ -37,14 +38,25 @@ QgsUserProfileSelectionDialog::QgsUserProfileSelectionDialog( QgsUserProfileMana
   // Add a new profile on button click
   connect( mAddProfileButton, &QPushButton::clicked, this, &QgsUserProfileSelectionDialog::onAddProfile );
 
+  // Remove selected profile on button click
+  connect( mRemoveProfileButton, &QPushButton::clicked, this, &QgsUserProfileSelectionDialog::onRemoveProfile );
+
   const int iconSize = mManager->settings()->value( QStringLiteral( "/selector/iconSize" ), 24 ).toInt();
   mProfileListWidget->setIconSize( QSize( iconSize, iconSize ) );
 
   // Fill the list of profiles
-  mProfileListWidget->clear();  // Clear bogus profiles in the Ui form
+  populateProfileList();
+}
+
+void QgsUserProfileSelectionDialog::populateProfileList()
+{
+  mProfileListWidget->clear();
   for ( auto profile : mManager->allProfiles() )
   {
     auto item = new QListWidgetItem( mManager->profileForName( profile )->icon(), profile );
+    if ( profile == mActiveProfile )
+      item->setBackground( Qt::lightGray );
```

Comment:
This won't play nice with systems using dark themes -- we need to instead get a color from the widget's palette().